dask_expr/_core.py

```diff
@@ -332,7 +341,8 @@ def simplify(self) -> Expr:
         expr = self
         while True:
             dependents = collect_dependents(expr)
-            new = expr.simplify_once(dependents=dependents)
+            simplified = {}
+            new = expr.simplify_once(dependents=dependents, simplified=simplified)
```

Comment:
Can you pass the dict in directly? I want to avoid that someone takes this and forgets to clear it on every pass down the line. We avoid this if we don't have a variable

Reply:
Yeah, good call. We can definitely do that.